src/IO/H5/VolumeData.cpp

```diff
@@ -154,6 +162,308 @@ void append_element_extents_and_connectivity(
     }
   }
 }
+// Given a std::vector of grid_names, computes the number of blocks that exist
+// and also returns a std::vector of block numbers that is a one-to-one mapping
+// to each element in grid_names
+std::tuple<size_t, std::vector<size_t>, std::vector<std::vector<size_t>>>
+compute_and_organize_block_info(const std::vector<std::string>& grid_names) {
+  std::vector<size_t> block_number_for_each_element;
+  std::vector<std::vector<size_t>> sorted_element_indices;
+  block_number_for_each_element.reserve(grid_names.size());
+
+  // Fills block_number_for_each_element
+  for (const std::string& grid_name : grid_names) {
+    size_t end_position = grid_name.find(',', 1);
+    block_number_for_each_element.push_back(
+        static_cast<size_t>(std::stoi(grid_name.substr(2, end_position))));
+  }
+
+  auto max_block_number =
+      *std::max_element(block_number_for_each_element.begin(),
+                        block_number_for_each_element.end());
+  auto number_of_blocks = max_block_number + 1;
+  sorted_element_indices.reserve(number_of_blocks);
```

Comment:
I also want to call out your use of `.reserve()`. That's great! Definitely the right thing to do :D